src/ontodia/data/sparql/sparqlDataProvider.ts

```diff
@@ -265,15 +310,61 @@ export class SparqlDataProvider implements DataProvider {
                 LIMIT ${params.limit} OFFSET ${params.offset}
             }
             ${resolveTemplate(this.settings.filterElementInfoPattern, {dataLabelProperty})}
+            OPTIONAL {
+                FILTER (ISBLANK(?inst)).
+                {
+                    ?inst ?blankTrgProp ?blankTrg.
+                    ?blankSrc ?blankSrcProp ?inst.
+                    FILTER NOT EXISTS { ?inst rdf:first _:smth1 }.
+                    BIND("blankNode" as ?blankType)
+                } UNION {
+                    ?inst rdf:rest*/rdf:first ?blankTrg.
+                    ?blankSrc ?blankSrcProp ?inst.
+                    _:smth2 rdf:first ?blankTrg.
+                    BIND(?blankSrcProp as ?blankTrgProp)
+                    BIND("listHead" as ?blankType)
+                    FILTER NOT EXISTS { _:smth3 rdf:rest ?inst }.
+                } UNION {
+                    ?listHead rdf:rest* ?inst.
+                    FILTER NOT EXISTS { _:smth4 rdf:rest ?listHead }.
+
+                    ?listHead rdf:rest*/rdf:first ?blankTrg.
+                    ?blankSrc ?blankSrcProp ?listHead.
+                    _:smth5 rdf:first ?blankTrg.
+                    BIND(?blankSrcProp as ?blankTrgProp)
+                    BIND("listHead" as ?blankType)
+                }
+            }
         } ${textSearchPart ? 'ORDER BY DESC(?score)' : ''}
         `;
 
-        return this.executeSparqlQuery<ElementBinding>(query).then(getFilteredData);
+        return this.executeSparqlQuery<ElementBinding | BlankBinding>(query, {
+            additionalData: filterData,
+        }).then(result => {
+            if (this.options.acceptBlankNodes) {
+                return BlankNodes.updateFilterResults(
+                    result,
+                    (callBackQuery: string) => {
+                        return this.executeSparqlQuery<BlankBinding>(callBackQuery);
+                    },
+                );
+            } else {
+                result.results.bindings = result.results.bindings.filter(b => !isBlankBinding(b));
```

Comment:
Same here: move `isBlankBinding` check into `getFilteredData()` instead of re-assigning bindings.